### PR TITLE
Add function to convert single channel image data to RGB image

### PR DESCRIPTION
**Pull request by iche033**
# 🎉 New feature

## Summary

Used by https://github.com/ignitionrobotics/ign-gui/pull/212 and https://github.com/ignitionrobotics/ign-launch/pull/112

Added a generic helper function for converting single channel image data, e.g. thermal and depth images, to an RGB image. This is intended to be used for visualization. Optional arguments can be specified to tweak the visualization. 

## Test it

The corresponding PRs in ign gui updates the Image Display to use this new function: https://github.com/ignitionrobotics/ign-gui/pull/212

You can test this together with that PR by running the `sensors_demo.sdf` world to see the visualization in Image Display

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: graphics/include/ignition/common/Image.hh [hunk 1]
```diff
@@ -185,6 +186,78 @@ namespace ignition
...
+        T min = std::numeric_limits<T>::max();
+        T max = std::numeric_limits<T>::lowest();
+        if (_min > max)
+        {
+          for (unsigned int i = 0; i < samples; ++i)
+          {
+            T v = static_cast<T>(buffer[i]);
+            if (v > max && !std::isinf(v))
+              max = v;
+            if (v < min && !std::isinf(v))
```
Comment (chapulina) [inline graphics/include/ignition/common/Image.hh:228]: Windows is not happy about these `isinf` calls with `uint8_t`:

<details><summary>Compiler error</summary>
<p>

```
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(409): error C2668: 'fpclassify': ambiguous call to overloaded function
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(300): note: could be 'int fpclassify(long double) noexcept'
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(295): note: or       'int fpclassify(double) noexcept'
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(290): note: or       'int fpclassify(float) noexcept'
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(409): note: while trying to match the argument list '(_Ty)'
        with
        [
            _Ty=uint8_t
        ]
C:\Jenkins\workspace\ignition_common-ci-pr_any-windows7-amd64\ws\ign-common\graphics\include\ignition/common/Image.hh(226): note: see reference to function template instantiation 'bool isinf<T>(_Ty) noexcept' being compiled
        with
        [
            T=uint8_t,
            _Ty=uint8_t
        ]
C:\Jenkins\workspace\ignition_common-ci-pr_any-windows7-amd64\ws\ign-common\graphics\src\Image_TEST.cc(351): note: see reference to function template instantiation 'void ignition::common::Image::ConvertToRGBImage<uint8_t>(const void *,unsigned int,unsigned int,ignition::common::Image &,T,T,bool)' being compiled
        with
        [
            T=uint8_t
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\xstring(1739): note: see reference to class template instantiation 'std::basic_string_view<char,std::char_traits<char>>' being compiled
```

</p>
</details>

File: graphics/include/ignition/common/Image.hh [hunk 2]
```diff
@@ -185,6 +186,81 @@ namespace ignition
...
+      /// when normalizing the input image data. For example, if min and max
+      /// are set to 1 and 10, a data value 2 will be normalized to:
+      ///    (2 - 1) / (10 - 1) * 255.
+      /// \param[in] _data input image data buffer
+      /// \param[in] _width image width
+      /// \param[in] _height image height
+      /// \param[out] _output Output RGB image
+      /// \param[in] _min Minimum value to be used when normalizing the input
+      /// image data to RGB.
+      /// \param[in] _min Maximum value to be used when normalizing the input
```
Comment (ahcorde) [inline graphics/include/ignition/common/Image.hh:201]:
```suggestion
      /// \param[in] _max Maximum value to be used when normalizing the input
```

File: graphics/include/ignition/common/Image.hh [hunk 3]
```diff
@@ -185,6 +186,81 @@ namespace ignition
...
+      /// image data to RGB.
+      /// \param[in[ _flip True to flip the values after normalization, i.e.
```
Comment (ahcorde) [inline graphics/include/ignition/common/Image.hh:203]:
```suggestion
      /// \param[in] _flip True to flip the values after normalization, i.e.
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: graphics/include/ignition/common/Image.hh (hunk 3 above)
Comment (iche033) [reply graphics/include/ignition/common/Image.hh]: fe9fffe6e07e66f4b28d96632ed02571e8591ddc

File: graphics/include/ignition/common/Image.hh (hunk 2 above)
Comment (iche033) [reply graphics/include/ignition/common/Image.hh]: fe9fffe6e07e66f4b28d96632ed02571e8591ddc

**Review by ahcorde — APPROVED** [review]

**Review by mjcarroll — COMMENTED** [review]
An approach to remove `new/delete` as well as the raw pointers: https://github.com/ignitionrobotics/ign-common/pull/206

**Review by mjcarroll — APPROVED** [review]
LGTM

**Pull request merged**